### DISCUSSION TITLE
按start程序直接关闭的问题

### DISCUSSION THREAD

**Discussion by Moon-Minotaur** [NateScarlet/auto-derby#176] [Q&A] [answered]
————launcher.log
**********************
Windows PowerShell 脚本开始
开始时间: 20210904022445
用户名: DESKTOP-V92I0SR\YUE
RunAs 用户: DESKTOP-V92I0SR\YUE
配置名称: 
计算机: DESKTOP-V92I0SR (Microsoft Windows NT 10.0.19042.0)
主机应用程序: PowerShell -Version 3 -NoProfile -ExecutionPolicy Bypass . "C:\HDDapps\auto-derby-master\auto_derby\launcher\launcher.ps1"
进程 ID: 2524
PSVersion: 5.1.19041.1151
PSEdition: Desktop
PSCompatibleVersions: 1.0, 2.0, 3.0, 4.0, 5.0, 5.1.19041.1151
BuildVersion: 10.0.19041.1151
CLRVersion: 4.0.30319.42000
WSManStackVersion: 3.0
PSRemotingProtocolVersion: 2.3
SerializationVersion: 1.1.0.1
**********************
已启动脚本，输出文件为 C:\HDDapps\auto-derby-master\launcher.log

Job                       : nurturing
Debug                     : True
CheckUpdate               : True
PythonExecutablePath      : C:\Program Files\Python38\pythonw.exe
SingleModeChoicesDataPath : C:\HDDapps\auto-derby-master\data\single_mode_choices.csv
PauseIfRaceOrderGt        : 10
Plugins                   :
TargetTrainingLevels      : 6,5,3,0,
ADBAddress                :
Version                   : 1.6.1
Python Version            :

command:
title auto-derby: 1.6.1
cd /d "C:\HDDapps\auto-derby-master"
set "DEBUG=auto_derby"
set "AUTO_DERBY_LAST_SCREENSHOT_SAVE_PATH=debug/last_screenshot.png"
set "AUTO_DERBY_OCR_IMAGE_PATH=debug/ocr_images"
set "AUTO_DERBY_SINGLE_MODE_EVENT_IMAGE_PATH=debug/single_mode_event_images"
set "AUTO_DERBY_SINGLE_MODE_TRAINING_IMAGE_PATH=debug/single_mode_training_images"
set "AUTO_DERBY_SINGLE_MODE_CHOICE_PATH=C:\HDDapps\auto-derby-master\data\single_mode_choices.csv"
set "AUTO_DERBY_PAUSE_IF_RACE_ORDER_GT=10"
set "AUTO_DERBY_PLUGINS="
set "AUTO_DERBY_SINGLE_MODE_TARGET_TRAINING_LEVELS=6,5,3,0,"
set "AUTO_DERBY_ADB_ADDRESS="
set "AUTO_DERBY_CHECK_UPDATE=true"
"C:\Program Files\Python38\pythonw.exe" -m auto_derby nurturing
exit

Installed packages:

adb-shell             0.3.3
cast-unknown          0.1.6
mouse                 0.7.1
numpy                 1.20.3
opencv-contrib-python 4.5.2.52
opencv-python         4.5.2.52
Pillow                8.2.0
pywin32               300

**********************
Windows PowerShell 脚本结束
结束时间: 20210904022454
**********************

————auto_derby.log
INFO  [2021-09-04 02:24:53]:__main__:23: auto_derby: 1.6.1 (2021-08-29 15:09:06)
DEBUG [2021-09-04 02:24:53]:auto_derby.plugin:43: loaded: afk, auto_crane, example_custom_context, example_custom_race_score, example_custom_training_partner_score, example_custom_training_score, example_power_level5, example_smart_falcon, hello_world, limited_sale_buy_everything, limited_sale_buy_first_3, limited_sale_ignore, limited_sale_close, more_g1, prefer_g1, no_event_prompt, no_ocr_prompt, pause_before_command, pause_before_race_continue, SSR駿川たづな, umapyoi, use_legacy_screenshot
INFO  [2021-09-04 02:24:53]:auto_derby.window:82: width=540 height=960
INFO  [2021-09-04 02:24:53]:auto_derby.window:84: already in wanted size
DEBUG [2021-09-04 02:24:53]:auto_derby.template:44: screenshot
DEBUG [2021-09-04 02:24:53]:auto_derby.template:58: load: connecting.pos.png
DEBUG [2021-09-04 02:24:53]:auto_derby.template:58: load: connecting.png
DEBUG [2021-09-04 02:24:53]:auto_derby.template:176: not match: tmpl=tmpl<connecting.png>, pos=(391, 55), similarity=-0.031
DEBUG [2021-09-04 02:24:53]:auto_derby.template:58: load: retry_button.pos.png
DEBUG [2021-09-04 02:24:53]:auto_derby.template:75: can not load: retry_button.pos.png: [Errno 2] No such file or directory: 'C:\\HDDapps\\auto-derby-master\\auto_derby\\templates\\retry_button.pos.png'
DEBUG [2021-09-04 02:24:53]:auto_derby.template:58: load: retry_button.png
DEBUG [2021-09-04 02:24:54]:auto_derby.template:176: not match: tmpl=tmpl<retry_button.png>, pos=(151, 925), similarity=0.379
DEBUG [2021-09-04 02:24:54]:auto_derby.template:58: load: single_mode_command_training.pos.png
DEBUG [2021-09-04 02:24:54]:auto_derby.template:58: load: single_mode_command_training.png
DEBUG [2021-09-04 02:24:54]:auto_derby.template:121: lightness match: tmpl=tmpl<single_mode_command_training.png>, similarity=1.000
INFO  [2021-09-04 02:24:54]:auto_derby.template:180: match: tmpl=tmpl<single_mode_command_training.png>, pos=(199, 760), similarity=0.97
INFO  [2021-09-04 02:24:54]:auto_derby.window:82: width=540 height=960
INFO  [2021-09-04 02:24:54]:auto_derby.window:84: already in wanted size
DEBUG [2021-09-04 02:24:54]:auto_derby.template:44: screenshot
DEBUG [2021-09-04 02:24:54]:auto_derby.template:58: load: single_mode_character_detail_button.pos.png
DEBUG [2021-09-04 02:24:54]:auto_derby.template:58: load: single_mode_character_detail_button.png
DEBUG [2021-09-04 02:24:54]:auto_derby.template:121: lightness match: tmpl=tmpl<single_mode_character_detail_button.png>, similarity=1.000
INFO  [2021-09-04 02:24:54]:auto_derby.template:180: match: tmpl=tmpl<single_mode_character_detail_button.png>, pos=(466, 559), similarity=0.95
DEBUG [2021-09-04 02:24:54]:auto_derby.ocr:151: match label: value=), current=0000f66ff76fe101e003a003a003a002a002a00bac1be813a812a8bbffff0000, match=7800f801f803e003c007c0078007800780078007c007c007e007f003f801f800, similarity=0.406
ERROR [2021-09-04 02:24:54]:__main__:123: unexpected exception
Traceback (most recent call last):
  File "C:\HDDapps\auto-derby-master\auto_derby\__main__.py", line 119, in <module>
    main()
  File "C:\HDDapps\auto-derby-master\auto_derby\__main__.py", line 90, in main
    job()
  File "C:\HDDapps\auto-derby-master\auto_derby\jobs\nurturing.py", line 100, in nurturing
    _handle_turn(ctx)
  File "C:\HDDapps\auto-derby-master\auto_derby\jobs\nurturing.py", line 36, in _handle_turn
    ctx.scene.recognize(ctx)
  File "C:\HDDapps\auto-derby-master\auto_derby\scenes\single_mode\command.py", line 82, in recognize
    ctx.update_by_command_scene(template.screenshot(max_age=0))
  File "C:\HDDapps\auto-derby-master\auto_derby\single_mode\context.py", line 284, in update_by_command_scene
    self.date = _ocr_date(screenshot.crop(date_bbox))
  File "C:\HDDapps\auto-derby-master\auto_derby\single_mode\context.py", line 51, in _ocr_date
    text = ocr.text(image_from_array(binary_img))
  File "C:\HDDapps\auto-derby-master\auto_derby\ocr.py", line 359, in text
    ret += _text_from_image(i, threshold)
  File "C:\HDDapps\auto-derby-master\auto_derby\ocr.py", line 160, in _text_from_image
    return _prompt(img, h, value, similarity)
  File "C:\HDDapps\auto-derby-master\auto_derby\ocr.py", line 130, in _prompt
    ans = terminal.prompt(
  File "C:\HDDapps\auto-derby-master\auto_derby\terminal.py", line 29, in prompt
    return input(message)
RuntimeError: input(): lost sys.stdin

————auto_derby.log.1
INFO  [2021-09-04 02:21:33]:__main__:23: auto_derby: 1.6.1 (2021-08-29 15:09:06)
DEBUG [2021-09-04 02:21:33]:auto_derby.plugin:43: loaded: afk, auto_crane, example_custom_context, example_custom_race_score, example_custom_training_partner_score, example_custom_training_score, example_power_level5, example_smart_falcon, hello_world, limited_sale_buy_everything, limited_sale_buy_first_3, limited_sale_ignore, limited_sale_close, more_g1, prefer_g1, no_event_prompt, no_ocr_prompt, pause_before_command, pause_before_race_continue, SSR駿川たづな, umapyoi, use_legacy_screenshot
INFO  [2021-09-04 02:21:33]:auto_derby.plugin:48: installed: no_ocr_prompt
INFO  [2021-09-04 02:21:33]:auto_derby.plugin:48: installed: no_event_prompt
INFO  [2021-09-04 02:21:33]:auto_derby.plugin:48: installed: auto_crane
INFO  [2021-09-04 02:21:33]:auto_derby.plugin:48: installed: afk
INFO  [2021-09-04 02:21:33]:auto_derby.window:82: width=559 height=994
INFO  [2021-09-04 02:21:33]:auto_derby.window:82: width=540 height=960
INFO  [2021-09-04 02:21:33]:auto_derby.window:84: already in wanted size
DEBUG [2021-09-04 02:21:34]:auto_derby.template:44: screenshot
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: connecting.pos.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: connecting.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:176: not match: tmpl=tmpl<connecting.png>, pos=(391, 55), similarity=-0.025
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: retry_button.pos.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:75: can not load: retry_button.pos.png: [Errno 2] No such file or directory: 'C:\\HDDapps\\auto-derby-master\\auto_derby\\templates\\retry_button.pos.png'
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: retry_button.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:176: not match: tmpl=tmpl<retry_button.png>, pos=(151, 925), similarity=0.389
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: single_mode_command_training.pos.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: single_mode_command_training.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:121: lightness match: tmpl=tmpl<single_mode_command_training.png>, similarity=1.000
INFO  [2021-09-04 02:21:34]:auto_derby.template:180: match: tmpl=tmpl<single_mode_command_training.png>, pos=(199, 760), similarity=0.97
INFO  [2021-09-04 02:21:34]:auto_derby.window:82: width=540 height=960
INFO  [2021-09-04 02:21:34]:auto_derby.window:84: already in wanted size
DEBUG [2021-09-04 02:21:34]:auto_derby.template:44: screenshot
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: single_mode_character_detail_button.pos.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:58: load: single_mode_character_detail_button.png
DEBUG [2021-09-04 02:21:34]:auto_derby.template:121: lightness match: tmpl=tmpl<single_mode_character_detail_button.png>, similarity=1.000
INFO  [2021-09-04 02:21:34]:auto_derby.template:180: match: tmpl=tmpl<single_mode_character_detail_button.png>, pos=(466, 559), similarity=0.95
DEBUG [2021-09-04 02:21:34]:auto_derby.ocr:151: match label: value=), current=0000f66ff76fe101e003a003a003a002a002a00bac1be813a812a8bbffff0000, match=7800f801f803e003c007c0078007800780078007c007c007e007f003f801f800, similarity=0.406
WARNING[2021-09-04 02:21:34]:auto_derby.ocr:116: using low similarity label: hash=0000f66ff76fe101e003a003a003a002a002a00bac1be813a812a8bbffff0000, value=), similarity=0.40625
DEBUG [2021-09-04 02:21:34]:auto_derby.ocr:151: match label: value=川, current=120036c036e086e787c79307130613061a0618061c060ce68eef86ef862f0200, match=18c018e718e718e718e718e718e718e71ce71ce71ce71ce70ee70ee70fe706c2, similarity=0.344
WARNING[2021-09-04 02:21:34]:auto_derby.ocr:116: using low similarity label: hash=120036c036e086e787c79307130613061a0618061c060ce68eef86ef862f0200, value=川, similarity=0.34375
DEBUG [2021-09-04 02:21:34]:auto_derby.ocr:361: ocr result: )川
ERROR [2021-09-04 02:21:34]:__main__:123: unexpected exception
Traceback (most recent call last):
  File "C:\HDDapps\auto-derby-master\auto_derby\__main__.py", line 119, in <module>
    main()
  File "C:\HDDapps\auto-derby-master\auto_derby\__main__.py", line 90, in main
    job()
  File "C:\HDDapps\auto-derby-master\auto_derby\jobs\nurturing.py", line 100, in nurturing
    _handle_turn(ctx)
  File "C:\HDDapps\auto-derby-master\auto_derby\jobs\nurturing.py", line 36, in _handle_turn
    ctx.scene.recognize(ctx)
  File "C:\HDDapps\auto-derby-master\auto_derby\scenes\single_mode\command.py", line 82, in recognize
    ctx.update_by_command_scene(template.screenshot(max_age=0))
  File "C:\HDDapps\auto-derby-master\auto_derby\single_mode\context.py", line 284, in update_by_command_scene
    self.date = _ocr_date(screenshot.crop(date_bbox))
  File "C:\HDDapps\auto-derby-master\auto_derby\single_mode\context.py", line 57, in _ocr_date
    year_end = text.index("級") + 1
ValueError: substring not found

**Comment by NateScarlet** [comment, marked as answer]
现在只支持第一个剧本 #172
如果你确定是第一个剧本 发 debug/last_screenshot.png

**Reply by Moon-Minotaur** [reply]
明白了，谢谢
